OSDOCS#7888: Document support for the enable-certificate-owner-ref flag (#65158)
openshift/openshift-docs: modules/cert-manager-override-flag-controller.adoc

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc

:_content-type: PROCEDURE
[id="cert-manager-override-flag-controller_{context}"]

> = Deleting TLS secret automatically upon Certificate removal

Reviewer: ```suggestion
= Deleting a TLS secret automatically upon Certificate removal
```

> = Customizing the cert-manager controller by overriding flags

Reviewer 2: ```suggestion
= Customizing cert-manager to automatically delete TLS secret when certificate resource is removed
```

This section seems bit repetitive as it also is part of spec.controllerConfig.overrideArgs which was in previous section as well. Since this is an important feature with a warning, IMHO should be fine to add a separate section but renaming it accordingly to reflect the use case of enable-certificate-owner-ref.

Author: Can we call it "Managing TLS secrets and certificates"

Author: The heading will be long otherwise. We can explain that cert-manager deletes the secret automatically  when certificate resource is removed in the paragraph.

> You can enable the `--enable-certificate-owner-ref` flag for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.

Reviewer: What does the `--enable-certificate-owner-ref` flag do?

Reviewer: I see the definition below. Maybe move the _ The `--enable-certificate-owner-ref` flag sets..._ sentence here? It might be more helpful earlier in the topic.

Author: Agree

.Prerequisites

* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
* You have installed the {cert-manager-operator}.

.Procedure

> . Check that the `ClusterIssuer` object and its secret is available by running the following command:

Reviewer 2: ```suggestion
. Check that the `Certificate` object and its secret is available by running the following command:
```

> . Check that the `Certificate` object and its secret is available by running the following command:

Reviewer: ```suggestion
. Check that the `Certificate` object and its secret are available by running the following command:
```

+
[source,terminal]
----

> $ oc get cert

Reviewer 2: Nit: can we use `oc get certificate`?

----
+
.Example output
[source,terminal]
----
NAME                                             READY   SECRET                                           AGE
certificate-from-clusterissuer-route53-ambient   True    certificate-from-clusterissuer-route53-ambient   8h
----

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager cluster
----

. Add a `spec.controllerConfig` section with the following override arguments:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
metadata:
  name: cluster

>   ...

Reviewer: ```suggestion
#  ...
```

spec:

>   ...

Reviewer: ```suggestion
#  ...
```

  controllerConfig:
    overrideArgs:
      - '--enable-certificate-owner-ref' <1>
----
<1> The `--enable-certificate-owner-ref` flag sets the certificate resource as an owner of the secret where the TLS certificate is stored.
+
[WARNING]
====

> If you uninstall the cert-manager Operator, cert-manager controller, CA injector, Webhook, or certificate resource, the secret is deleted automatically. This might cause network connectivity issues.

Reviewer 2: ```suggestion
If you uninstall the {cert-manager-operator} or delete the certificate resource(s) from the cluster, the secret is deleted automatically. This might cause network connectivity issues depending upon where the certificate TLS secret is being used.
```

====

. Save your changes and quit the text editor to apply your changes.

.Verification

* Verify that the `--enable-certificate-owner-ref` flag is updated for cert-manager controller pod by running the following command:
+
[source,terminal]
----
$ oc get pods -l app.kubernetes.io/name=cert-manager -n cert-manager -o yaml
----
+
.Example output
[source,yaml]
----

> ...

Reviewer: ```suggestion
# ...
```

  metadata:
    name: cert-manager-6e4b4d7d97-zmdnb
    namespace: cert-manager

> ...

Reviewer: ```suggestion
# ...
```
